Added lifecycle policy

ivan-pinatti opened the pull request:
## what
* Define an `aws_ecr_lifecycle_policy` to expunge old images

## why
* This is a draft version that solves the issue [https://github.com/cloudposse/jenkins/issues/8](https://github.com/cloudposse/jenkins/issues/8).

I do not know your coding patterns, I assumed only untagged for now. However, you can add a ternary to select between tagged/untagged. 

Cheers!

aknysh commented on variables.tf:
```diff
@@ -24,3 +24,9 @@ variable "tags" {
   default = {}
 }
+
+variable "rotate_number" {
```
would be better to name it `max_image_number`

aknysh commented:
please `terraform fmt` all the code

```
$ make terraform:lint
Terraform configuration needs linting. Run '/home/travis/build/cloudposse/terraform-aws-ecr/build-harness/vendor/terraform fmt'
- main.tf - variables.tf
make: *** [terraform:lint] Error 1
The command "make terraform:lint" exited with 2.
```

ivan-pinatti commented:
@aknysh,

Yeap, it worked.
Output was;

```
Apply complete! Resources: 8 added, 0 changed, 0 destroyed.

Outputs:

registry_id = 111111111111
registry_url = 111111111111.dkr.ecr.us-east-1.amazonaws.com/test-prod-test
repository_name = test-prod-test
role_name = test-prod-test
```

In AWS console;

![image](https://user-images.githubusercontent.com/7431771/35576678-959ea624-05ad-11e8-8aae-bcdbdf69e606.png)

![image](https://user-images.githubusercontent.com/7431771/35576687-9e6acad0-05ad-11e8-9bb8-7a74045869e5.png)

Already renamed the variables. :)

Cheers!

aknysh commented:
@ivan-pinatti 
great, thanks for testing
we are almost there :)
can you rename the var to `max_image_count` and add the var to `README`?
https://github.com/cloudposse/terraform-aws-ecr/blob/master/README.md#variables

ivan-pinatti commented:
@aknysh,

I've renamed and updated the README file, however, I changed the logic to be tagged by default (which makes more sense).

I couldn't implement the ternary + variable to select, because Terraform cannot output lists from the ternary yet, there is an open issue about it, [https://github.com/hashicorp/terraform/issues/12453#issuecomment-284273475](https://github.com/hashicorp/terraform/issues/12453#issuecomment-284273475).
I suggest to add it in a TO-DO list for when it is fixed.

Because it is now assuming by default that it **tagged**, I will drop a PR later in the other repo to tag the image with the stage name as well.

ivan-pinatti commented:
@osterman,

My two cents, create a project (kanban style) in this repo and add a TO-DO column to keep track of issues that we already know, like the ternary that cannot output lists and block us to implement the selection (tagged|untagged).
